data/food/export_agb/export_builder.py

```diff
@@ -212,18 +244,32 @@ def compute_new_processes(activities, ingredients_base):
             v["export_data"]["impacts"]["bvi"] = 0.0
 
     # Compute new processes for complex variants
-    new_processes = compute_new_processes(activities, ingredients_base)
+    new_processes = compute_new_processes(activities, ingredients)
 
     # Export the ingredients
-    print(f"Export de {len(ingredients_base)} ingrédients vers {INGREDIENTS}")
+    print(f"Export de {len(ingredients)} ingrédients vers {INGREDIENTS}")
     with open(INGREDIENTS, "w") as outfile:
-        json.dump(ingredients_base, outfile, indent=2, ensure_ascii=False)
-        outfile.write("\n")  # Add a newline at the end of the file, as many editors do.
+        json.dump(ingredients, outfile, indent=2, ensure_ascii=False)
 
     # Add the new processes we computed for the complex ingredients
     export = [v["export_data"] for v in activities.values()] + new_processes
 
+    # compute the corrected impacts
+    print("Computing corrected impacts from weihgted subimpacts")
+    corrections = {
+        k: v["correction"] for (k, v) in impacts_ecobalyse.items() if "correction" in v
+    }
+    for process in export:
+        for impact_to_correct, correction in corrections.items():
+            corrected_impact = 0
+            for correction_item in correction:  # For each sub-impact and its weighting
+                sub_impact_name = correction_item["sub-impact"]
+                if sub_impact_name in process["impacts"]:
+                    sub_impact = process["impacts"].get(sub_impact_name, 1)
+                    corrected_impact += sub_impact * correction_item["weighting"]
+                    del process["impacts"][sub_impact_name]
+            process["impacts"][impact_to_correct] = corrected_impact
+
     print(f"Export de {len(export)} procédés vers {BUILDER}")
     with open(BUILDER, "w") as outfile:
         json.dump(export, outfile, indent=2, ensure_ascii=False)
-        outfile.write("\n")  # Add a newline at the end of the file, as many editors do.
```

Comment:
I believe @n1k0 asked me to do this, so when you open the exported file in an editor that adds a newline at the end of the file, it doesn't create a diff, so maybe it's better to keep it?

Reply:
Yes, please (thanks) 🙏